Sidebar TimeInfo 컴포넌트 제작

pieisland opened the pull request:
## 📕 제목

Sidebar TimeInfo 컴포넌트 제작
관련이슈 #6

## 📗 작업 내용

> 구현 내용 및 작업 했던 내역

- [x] timeinfo 컴포넌트를 제작했습니다.
- [x] sidebar에 time-info를 추가했습니다.
- [x] play time, total time, cursor time을 00.00.000 형식으로 나타냈습니다.

![image](https://user-images.githubusercontent.com/35261724/100070514-5eba7600-2e7d-11eb-93cb-fbe58ae60571.png)


## 📘 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 값을 string으로 넣어두었습니다.

pieisland committed:
<feat>: [FE] Sidebar 컴포넌트

- sidebar 컴포넌트를 제작했습니다.
- source list와 재생시간 정보, 단축키 정보 등이 들어가는 부분입니다.

pieisland committed:
<style>: [FE] Sidebar 컴포넌트

- sidebar 컴포넌트의 스타일을 지정했습니다.

pieisland committed:
<feat>: [FE] SourceList 컴포넌트

- sourcelist 컴포넌트를 제작했습니다.
- 개별 source 에 마우스를 올릴 시 정보가 나타납니다.

pieisland committed:
style: [FE] SourceList 컴포넌트

- sourcelist 컴포넌트의 스타일을 지정했습니다.

pieisland committed:
feat: [FE] TimeInfo 컴포넌트 제작

- timeinfo 컴포넌트를 제작했습니다.
- playTime, totalTime, cursorTime을 00.00.000 으로 지정해두었습니다.
  - string으로 하여 변환이 필요할 것으로 생각됩니다.

pieisland committed:
style: [FE] TimeInfo 컴포넌트 제작

- timeinfo 컴포넌트의 style을 지정했습니다.

pieisland committed:
style: [FE] Sidebar style 수정

- sidebar에 timeinfo를 추가하면서 css를 수정했습니다.

Jeongeun-Choi commented on frontend/src/components/sidebar/sidebar.scss:
```diff
@@ -0,0 +1,23 @@
+#sidebar {
+    border:1px solid white;
+    background-color: black;
+    width:20%;
+    height:80%;
+    position:absolute;
+    padding:1px;
+
+    display:flex;
+    flex-direction:column;
+    align-items:center;
+}
+
+.sidebar-child {
+    width:100%;
+}
+
+.sidebar-source-list {
+    height:50%;
+}
+.sidebar-time-info {
+    height:20%;
+}
```
파일 끝에는 항상 빈줄이 있는게 좋다고 합니다!😀
참고 : https://velog.io/@doondoony/posix-eol

pieisland replied:
그렇군요! 알려주셔서 감사합니다~

Jeongeun-Choi commented on frontend/src/components/source-list/index.ts:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    render() {
+      const sources = [{title:"sample1.mp4"}, {title:"sample2.mp4"}];
+      let sourceList:string="";
+      sources.forEach((source)=>{
+        sourceList += `<div class="audio-source">${source.title}
+                            <div class="source-info">정보...</div>
+                       </div>`;
+        } )
```
forEach를 사용하여 하나의 문자열로 만든거 정말 좋아요! 😊👍

Jeongeun-Choi reviewed (approved):
scss 파일들에서 파일 끝에 빈줄이 없는데 추가해주시면 좋을 것 같아요! 
코드 깔끔하고 좋습니다!😊

Songwonseok commented on frontend/src/components/source-list/index.ts:
```diff
@@ -0,0 +1,37 @@
+      let sourceList:string="";
+      sources.forEach((source)=>{
+        sourceList += `<div class="audio-source">${source.title}
+                            <div class="source-info">정보...</div>
```
reduce로 사용해도 좋을 것 같아요!

pieisland replied:
네 바꿔보겠습니다~

wooojini commented on frontend/src/components/sidebar/sidebar.scss:
```diff
@@ -0,0 +1,23 @@
+#sidebar {
+    border:1px solid white;
+    background-color: black;
```
CSS 정렬을 하면 어떨까요??? 이전에 리뷰어님께 리뷰받았던 사항이었는데, 아래 링크를 참조해주세요 :-)
참조 : https://webdesign.tutsplus.com/articles/outside-in-ordering-css-properties-by-importance--cms-21685

pieisland replied:
오 감사합니다~ 보고 정렬해보겠습니다.

wooojini commented on frontend/src/components/source-list/index.ts:
```diff
@@ -0,0 +1,37 @@
+                       </div>`;
+        } )
+
+      this.innerHTML = `
```
source들을 멤버변수로 선언하고, sourceList를 문자열로 concat하는 부분은 별도의 함수로 분리하는 것은 어떨까요?? 지금은 로직이 간단하지만, 이후에 로직이 복잡해질 수록 render함수에서 많은일을 수행할 것 같습니다! :-)

pieisland replied:
아 네 그러는 게 좋은 것 같네요. 별도의 함수로 분리해놓겠습니다!

wooojini reviewed (approved):
고생하셨습니다!! 전체적으로 깔끔하네요 :-) 리뷰 내용은 코멘트 남겼습니다!

pieisland committed:
refactor: [FE] scss 파일 이름 변경 및 스타일 정렬

- scss 파일 이름을 모두 style.scss로 변경했습니다.
- 스타일 정렬을 했습니다.

pieisland committed:
refactor: [FE] 스타일 import 이름 변경과 리뷰 적용

- scss 파일 이름을 바꾸면서 import 하는 부분을 변경했습니다.
- 리뷰 내용을 적용했습니다.
  - sourceList를 하나의 string으로 묶는 부분을 별도의 함수로 분리
  - forEach 대신 reducer 사용

wooojini committed:
Merge branch 'develop' into feature/sidebar-timeinfo
